module/x/gravity/keeper/keeper_valset.go

```diff
@@ -312,6 +312,13 @@ func (k Keeper) GetCurrentValset(ctx sdk.Context) (types.Valset, error) {
 	if err != nil {
 		return types.Valset{}, (sdkerrors.Wrap(err, types.ErrInvalidValset.Error()))
 	}
+
+	// this part is important for gravity contract
+	// since we expect the valset to be sorted by Eth Address in ASC order
+	sort.Slice(valset.Members, func(i, j int) bool {
```

Comment:
Update the sorting:

```
	// this part is important for gravity contract
	// since we expect the valset to be sorted by Eth Address in ASC order
	sort.Slice(valset.Members, func(i, j int) bool {
		return strings.ToLower(valset.Members[i].EthereumAddress) < strings.ToLower(valset.Members[j].EthereumAddress)
	})

```

Also you will need to add strings to imports
```
"fmt"
"math/big"
"sort"
"strconv"
"strings" <--- here

"github.com/cosmos/cosmos-sdk/store/prefix"	
```